[
    {
        "url": "https://api.github.com/repos/ThomasWaldmann/argparse/issues/100",
        "repository_url": "https://api.github.com/repos/ThomasWaldmann/argparse",
        "labels_url": "https://api.github.com/repos/ThomasWaldmann/argparse/issues/100/labels{/name}",
        "comments_url": "https://api.github.com/repos/ThomasWaldmann/argparse/issues/100/comments",
        "events_url": "https://api.github.com/repos/ThomasWaldmann/argparse/issues/100/events",
        "html_url": "https://github.com/ThomasWaldmann/argparse/issues/100",
        "id": 358309016,
        "node_id": "MDU6SXNzdWUzNTgzMDkwMTY=",
        "number": 100,
        "title": "Poll: why are you using argparse from pypi (or from here) these days?",
        "user": {
            "login": "ThomasWaldmann",
            "id": 356103,
            "node_id": "MDQ6VXNlcjM1NjEwMw==",
            "avatar_url": "https://avatars.githubusercontent.com/u/356103?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/ThomasWaldmann",
            "html_url": "https://github.com/ThomasWaldmann",
            "followers_url": "https://api.github.com/users/ThomasWaldmann/followers",
            "following_url": "https://api.github.com/users/ThomasWaldmann/following{/other_user}",
            "gists_url": "https://api.github.com/users/ThomasWaldmann/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/ThomasWaldmann/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/ThomasWaldmann/subscriptions",
            "organizations_url": "https://api.github.com/users/ThomasWaldmann/orgs",
            "repos_url": "https://api.github.com/users/ThomasWaldmann/repos",
            "events_url": "https://api.github.com/users/ThomasWaldmann/events{/privacy}",
            "received_events_url": "https://api.github.com/users/ThomasWaldmann/received_events",
            "type": "User",
            "site_admin": false
        },
        "labels": [],
        "state": "open",
        "locked": false,
        "assignee": null,
        "assignees": [],
        "milestone": null,
        "comments": 2,
        "created_at": "2018-09-08T16:25:16Z",
        "updated_at": "2018-11-28T23:01:41Z",
        "closed_at": null,
        "author_association": "OWNER",
        "active_lock_reason": null,
        "body": "Just wondering:\r\n\r\nConsidering that everybody should use py27 or py34+ these days (which all have argparse built-in), I am wondering why people keep starring this repository on github.\r\n\r\nDo you still use this? Why? Please leave comments below.\r\n\r\nSome stats: https://libraries.io/pypi/argparse\r\n",
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/ThomasWaldmann/argparse/issues/95",
        "repository_url": "https://api.github.com/repos/ThomasWaldmann/argparse",
        "labels_url": "https://api.github.com/repos/ThomasWaldmann/argparse/issues/95/labels{/name}",
        "comments_url": "https://api.github.com/repos/ThomasWaldmann/argparse/issues/95/comments",
        "events_url": "https://api.github.com/repos/ThomasWaldmann/argparse/issues/95/events",
        "html_url": "https://github.com/ThomasWaldmann/argparse/issues/95",
        "id": 134804433,
        "node_id": "MDU6SXNzdWUxMzQ4MDQ0MzM=",
        "number": 95,
        "title": "Issue with argument_group and required add_mutually_exclusive_group",
        "user": {
            "login": "pneff",
            "id": 7936,
            "node_id": "MDQ6VXNlcjc5MzY=",
            "avatar_url": "https://avatars.githubusercontent.com/u/7936?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/pneff",
            "html_url": "https://github.com/pneff",
            "followers_url": "https://api.github.com/users/pneff/followers",
            "following_url": "https://api.github.com/users/pneff/following{/other_user}",
            "gists_url": "https://api.github.com/users/pneff/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/pneff/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/pneff/subscriptions",
            "organizations_url": "https://api.github.com/users/pneff/orgs",
            "repos_url": "https://api.github.com/users/pneff/repos",
            "events_url": "https://api.github.com/users/pneff/events{/privacy}",
            "received_events_url": "https://api.github.com/users/pneff/received_events",
            "type": "User",
            "site_admin": false
        },
        "labels": [],
        "state": "open",
        "locked": false,
        "assignee": null,
        "assignees": [],
        "milestone": null,
        "comments": 1,
        "created_at": "2016-02-19T08:29:36Z",
        "updated_at": "2017-06-08T12:21:15Z",
        "closed_at": null,
        "author_association": "NONE",
        "active_lock_reason": null,
        "body": "The snippet below does behave differently with the PyPI version than it does with any of the built-in versions from Python 2.7+.\n\n``` python\nfrom __future__ import print_function\n\nimport argparse\n\nparser = argparse.ArgumentParser()\n\ngroup = parser.add_argument_group('Source arguments')\n# Add mutually exclusive group to the argument group.\n# The problem only appears in that combination. When using\n# a mutually exclusive group without an argument group, then\n# the code works consistently across all argparse versions.\noneof = group.add_mutually_exclusive_group(required=True)\noneof.add_argument('--option1')\noneof.add_argument('--option2')\n\nprint(parser.parse_args())\n```\n\nTest output from the various Python versions (all on Windows)\n#### Python 2.7 with Pypi package\n- Inside a virtualenv\n- Tried 1.1, 1.3 and 1.4 all with the same result\n- Note the absence of the validation output, the arguments are parsed\n\n```\npython parse_test.py\nNamespace(option1=None, option2=None)\n```\n#### Python 2.7\n\n```\nc:\\python27\\python .\\parse_test.py\nusage: parse_test.py [-h] (--option1 OPTION1 | --option2 OPTION2)\nparse_test.py: error: one of the arguments --option1 --option2 is required\n```\n#### Python 3.4\n\n```\nC:\\Python34\\python .\\parse_test.py\nusage: parse_test.py [-h] (--option1 OPTION1 | --option2 OPTION2)\nparse_test.py: error: one of the arguments --option1 --option2 is required\n```\n#### Python 2.5\n\n```\nC:\\Python35\\python .\\parse_test.py\nusage: parse_test.py [-h] (--option1 OPTION1 | --option2 OPTION2)\nparse_test.py: error: one of the arguments --option1 --option2 is required\n```\n",
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/ThomasWaldmann/argparse/issues/81",
        "repository_url": "https://api.github.com/repos/ThomasWaldmann/argparse",
        "labels_url": "https://api.github.com/repos/ThomasWaldmann/argparse/issues/81/labels{/name}",
        "comments_url": "https://api.github.com/repos/ThomasWaldmann/argparse/issues/81/comments",
        "events_url": "https://api.github.com/repos/ThomasWaldmann/argparse/issues/81/events",
        "html_url": "https://github.com/ThomasWaldmann/argparse/issues/81",
        "id": 106177311,
        "node_id": "MDU6SXNzdWUxMDYxNzczMTE=",
        "number": 81,
        "title": "About this issue tracker",
        "user": {
            "login": "GoogleCodeExporter",
            "id": 9614759,
            "node_id": "MDQ6VXNlcjk2MTQ3NTk=",
            "avatar_url": "https://avatars.githubusercontent.com/u/9614759?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/GoogleCodeExporter",
            "html_url": "https://github.com/GoogleCodeExporter",
            "followers_url": "https://api.github.com/users/GoogleCodeExporter/followers",
            "following_url": "https://api.github.com/users/GoogleCodeExporter/following{/other_user}",
            "gists_url": "https://api.github.com/users/GoogleCodeExporter/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/GoogleCodeExporter/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/GoogleCodeExporter/subscriptions",
            "organizations_url": "https://api.github.com/users/GoogleCodeExporter/orgs",
            "repos_url": "https://api.github.com/users/GoogleCodeExporter/repos",
            "events_url": "https://api.github.com/users/GoogleCodeExporter/events{/privacy}",
            "received_events_url": "https://api.github.com/users/GoogleCodeExporter/received_events",
            "type": "User",
            "site_admin": false
        },
        "labels": [
            {
                "id": 259582618,
                "node_id": "MDU6TGFiZWwyNTk1ODI2MTg=",
                "url": "https://api.github.com/repos/ThomasWaldmann/argparse/labels/wontfix",
                "name": "wontfix",
                "color": "ffffff",
                "default": true,
                "description": null
            }
        ],
        "state": "open",
        "locked": false,
        "assignee": null,
        "assignees": [],
        "milestone": null,
        "comments": 2,
        "created_at": "2015-09-12T18:23:51Z",
        "updated_at": "2015-10-22T18:33:31Z",
        "closed_at": null,
        "author_association": "NONE",
        "active_lock_reason": null,
        "body": "```\nDo not file issues here that can be reproduced with the argparse module from \nPython >= 2.7 standard library! Please file these on the python.org issue \ntracker.\n\nThe issue tracker here is for issues that ONLY reproduce when using the \nargparse package downloaded from here or pypi.\n\n```\n\nOriginal issue reported on code.google.com by `Thomas.J...@gmail.com` on 27 Mar 2011 at 1:17\n",
        "performed_via_github_app": null
    }
]